server/py/services/api/crud/model_monitoring/model_endpoints.py

```diff
@@ -102,60 +101,41 @@ async def create_model_endpoint(
-                )
-                model_endpoint.spec.function_uid = current_function.get(
-                    "metadata", {}
-                ).get("uid")
-            except mlrun.errors.MLRunNotFoundError:
-                logger.info("The model endpoint is created on a non-existing function")
-        model_obj = None
+        model_obj, model_uri = None, None
+        model_path = model_endpoint.spec.model_path
         if model_path and mlrun.datastore.is_store_uri(model_path):
             _, model_uri = mlrun.datastore.parse_store_uri(model_path)
             project, key, iteration, tag, tree, uid = parse_artifact_uri(
                 model_uri, model_endpoint.metadata.project
             )
-        else:
-            project, key, iteration, tag, tree, uid = (
-                model_endpoint.metadata.project,
-                model_endpoint.spec.model_db_key,
-                None,
-                model_endpoint.spec.model_tag,
-                None,
-                model_endpoint.spec.model_uid,
-            )
-        try:
-            logger.info("Getting model object from db")
-            model_obj = mlrun.artifacts.dict_to_artifact(
-                services.api.crud.Artifacts().get_artifact(
+            try:
+                logger.info("Getting model object from db")
+                # Retrieve the model object from the database to extract its ID.
+                # The ID is later used to link the model endpoint to the model object.
+                # Fetching it here prevents retrieving the model object twice.
+                db_artifact = framework.utils.singletons.db.get_db().read_artifact(
                     db_session,
-                    key=key,
-                    tag=tag,
-                    iter=iteration,
-                    project=project,
-                    producer_id=tree,
-                    object_uid=uid,
+                    key,
+                    tag,
+                    iteration,
+                    project,
+                    tree,
+                    uid,
+                    as_record=True,
                 )
-            )
-
-            model_endpoint.spec.model_name = model_obj.metadata.key
-            model_endpoint.spec.model_db_key = model_obj.spec.db_key
-            model_endpoint.spec.model_uid = model_obj.metadata.uid
-            model_endpoint.spec.model_tag = model_obj.tag
-            model_endpoint.metadata.labels.update(
-                model_obj.labels
-            )  # todo : check if we still need this
-        except mlrun.errors.MLRunNotFoundError:
-            logger.info("The model endpoint is created on a non-existing model")
+                artifact = db_artifact.full_object
+                model_obj = mlrun.artifacts.dict_to_artifact(
+                    mlrun.common.formatters.ArtifactFormat.format_obj(artifact, "full")
+                )
+                model_endpoint.spec._model_id = db_artifact.id
```

Comment:
where is the comment you promised ? 😂

Reply:
https://github.com/mlrun/mlrun/pull/7341/files/3d094e23ea871d7e4d1a2cd100d4c764732dec33#diff-a2c737ab2ec00c6e09dadbaf6862ea92fb172dfcaca18bfc67d3c279369f5095R113 :)